### PR TITLE
Add default value test for project properties

### PR DESCRIPTION
**Pull request by tomasdavidorg**

### CODE REVIEW

File: guvnor-project/guvnor-project-api/src/test/java/org/guvnor/common/services/project/preferences/GAVPreferencesTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Test
+    public void defaultValueTest() {
+        GAVPreferences gavPreferences = new GAVPreferences();
+        gavPreferences.defaultValue(gavPreferences);
+
+        assertFalse(gavPreferences.isChildGAVEditEnabled());
+        assertFalse(gavPreferences.isConflictingGAVCheckDisabled());
+
+        System.setProperty(GAVPreferences.CONFLICTING_GAV_CHECK_DISABLED, "true");
```
Comment (manstis) [inline guvnor-project/guvnor-project-api/src/test/java/org/guvnor/common/services/project/preferences/GAVPreferencesTest.java]: If you're setting System Properties in tests it's a good idea to restore them to their original values afterwards.

Comment (tomasdavidorg) [inline guvnor-project/guvnor-project-api/src/test/java/org/guvnor/common/services/project/preferences/GAVPreferencesTest.java]: Fixed. Thanks.

**Comment by manstis** [comment]
Thanks @tomasdavidorg

**Comment by manstis** [comment]
@tomasdavidorg Hi, unfortunately this change appears to have broken SuperDevMode:
```
 [ERROR] Errors in 'org/guvnor/common/services/project/preferences/GAVPreferences.java'
            [ERROR] Line 42: Only string constants may be used as property name in System.getProperty()
         [ERROR] Errors in 'org/guvnor/common/services/project/preferences/GAVPreferences.java'
            [ERROR] Line 47: Only string constants may be used as property name in System.getProperty()
```
Looks like we have to revert back to the literals.

Could you please submit another PR?

**Comment by manstis** [comment]
@tomasdavidorg I submitted a fix [here](https://github.com/droolsjbpm/guvnor/pull/396)

**Comment by tomasdavidorg** [comment]
@manstis Should I change the test? Or it is ok.

**Comment by manstis** [comment]
@tomasdavidorg Hi, the test is OK (the constant makes things easier, it's just a shame GWT didn't like them)

**Pull request merged**